x-pack/plugins/canvas/public/apps/workpad/workpad_app/workpad_telemetry.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { connect } from 'react-redux';
+// @ts-ignore
+import { trackCanvasUiMetric } from '../../../lib/ui_metric';
+// @ts-ignore
+import { getElementCounts } from '../../../state/selectors/workpad';
+// @ts-ignore
+import { getArgs } from '../../../state/selectors/resolved_args';
+
+const WorkpadLoadedMetric = 'workpad-loaded';
+const WorkpadLoadedWithErrorsMetric = 'workpad-loaded-with-errors';
+
+export { WorkpadLoadedMetric, WorkpadLoadedWithErrorsMetric };
+
+const mapStateToProps = (state: any) => ({
+  telemetryElementCounts: getElementCounts(state),
+  telemetryResolvedArgs: getArgs(state),
+});
+
+interface ElementCounts {
+  error: number;
+  pending: number;
+  ready: number;
+}
+
+interface WorkpadPage {
+  elements: Array<{ id: string }>;
+}
+
+interface Workpad {
+  pages: WorkpadPage[];
+  id: string;
+}
+
+interface ResolvedArgs {
+  [keys: string]: any;
+}
+
+interface ElementsLoadedTelemetryProps {
+  telemetryElementCounts: ElementCounts;
+  workpad: Workpad;
+  telemetryResolvedArgs: {};
+}
+
+function areAllElementsInResolvedArgs(workpad: Workpad, resolvedArgs: ResolvedArgs) {
+  const resolvedArgsElements = new Set(Object.keys(resolvedArgs));
```

Comment:
Is there an advantage to using a Set here and `resolvedArgsElements.has`, versus just using the array of keys and `resolvedArgsElements.includes` instead?

Reply:
Probably not. I think I was originally going to do a Set v Set comparison of element Ids from workpad and resolvedArgs, but was worried there could be a scenario where ResolvedArgs had additional args hanging around in it from a previous workpad.  So I'll switch back to just doing the array check.